### PR TITLE
MAILBOX-359 FlagsUpdated Scala Event

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

File: mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala [hunk 1]
```diff
@@ -91,6 +92,17 @@ private object DTO {
...
+
+  case class FlagsUpdated(sessionId: SessionId, user: User, path: MailboxPath, mailboxId: MailboxId, uids: List[MessageUid],
+                          updatedFlags: List[UpdatedFlags]) extends Event {
+    override def toJava: JavaEvent = new JavaFlagsUpdated(
+      sessionId,
+      user,
+      path.toJava,
+      mailboxId,
+      uids.asJava,
+      updatedFlags.map(updatedFlag => updatedFlag.toJava).asJava)
```
Comment (chibenwa) [inline mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala]: Does `.map(_.toJava)` works?

File: mailbox/event/json/src/main/scala/org/apache/james/event/json/UpdatedFlagsDTO.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  * under the License.                                           *
+  * ***************************************************************/
+
+package org.apache.james.event.json
+
+import org.apache.james.event.json.MetaDataDTO.Flags
+import org.apache.james.mailbox.MessageUid
+import org.apache.james.mailbox.model.{UpdatedFlags => JavaUpdatedFlags}
+
+object UpdatedFlagsDTO {
```
Comment (chibenwa) [inline mailbox/event/json/src/main/scala/org/apache/james/event/json/UpdatedFlagsDTO.scala]: Can we get this in a single `DTOs.scala` file?

Comment (trantienduchn) [reply mailbox/event/json/src/main/scala/org/apache/james/event/json/UpdatedFlagsDTO.scala]: will be...

File: mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala [hunk 3]
```diff
@@ -80,6 +83,17 @@ private object DTO {
...
   }
+
+  case class FlagsUpdated(sessionId: SessionId, user: User, path: MailboxPath, mailboxId: MailboxId, uids: List[MessageUid],
+                          updatedFlags: List[DTOs.UpdatedFlags]) extends Event {
+    override def toJava: JavaEvent = new JavaFlagsUpdated(
+      sessionId,
+      user,
+      path.toJava,
+      mailboxId,
+      uids.asJava,
```
Comment (chibenwa) [inline mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala]: We already have the list of UIDs as part of the updated flags list.

Does it makes sens to send the uid list two times in the JSON?

I do think that 

```
    private static final String DEFAULT_EVENT_JSON =
        "{" +
        "  \"FlagsUpdated\": {" +
        "    \"path\": {" +
        "      \"namespace\": \"#private\"," +
        "      \"user\": \"user\"," +
        "      \"name\": \"mailboxName\"" +
        "    }," +
        "    \"mailboxId\": \"18\"," +
//        "    \"uids\": [123456, 654321]," + // can be removed as the information is in updatedFlags
        "    \"sessionId\": 42," +
        "    \"updatedFlags\": [" +
        "      {" +
        "        \"uid\": 123456," +
        "        \"modSeq\": 35," +
        "        \"oldFlags\": [\"Old Flag 1\", \"\\\\Deleted\", \"\\\\Seen\"]," +
        "        \"newFlags\": [\"New Flag 1\", \"\\\\Answered\", \"\\\\Draft\"]" +
        "      }," +
        "      {" +
        "        \"uid\": 654321," +
        "        \"modSeq\": 36," +
        "        \"oldFlags\": [\"Old Flag 2\", \"\\\\Flagged\", \"\\\\Recent\"]," +
        "        \"newFlags\": [\"New Flag 2\", \"\\\\Answered\", \"\\\\Seen\" ]" +
        "      }" +
        "    ]," +
        "    \"user\": \"user\"" +
        "  }" +
        "}";
```

Comment (trantienduchn) [reply mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala]: If the content of `uids` and list of uids in `UpdatedFlags` are the same, Then is it legal to refactor `EventFactory` ? I see factory method carries both list of uids and list of updatedFlags.

Comment (chibenwa) [reply mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala]: Maybe `FlagsUpdatedEvent` should just tale a List<UpdatedFlags> and compute `List<MessageUid>` from it, don't you think?

Comment (trantienduchn) [reply mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala]: I will do it in separated PR

Comment (trantienduchn) [reply mailbox/event/json/src/main/scala/org/apache/james/event/json/EventSerializer.scala]: #2055

**Comment by trantienduchn** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[a26bc7f8c46336717333d7d6f439e569e07c3148] [ERROR] Failures: 
[a26bc7f8c46336717333d7d6f439e569e07c3148] [ERROR]   CassandraMailboxRecentDAOTest.getRecentMessageUidsInMailboxShouldNotTimeoutWhenOverPagingLimit:123 
[a26bc7f8c46336717333d7d6f439e569e07c3148] Expected size:<6000> but was:<7000> in:
```
test this please

**Comment by trantienduchn** [comment]
test this please. rabbitmq forward test timeout

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**